pkgs/tools/package-management/wapm/cli/default.nix

```diff
@@ -0,0 +1,52 @@
+{ darwin
+, fetchFromGitHub
+, lib
+, libiconv
+, openssl
+, rustPlatform
+, stdenv }:
+
+let
+  # Constants
+  pname = "wapm-cli";
+  owner = "wasmerio";
+  repo = "wapm-cli";
+
+  # Per-version variables
+  version = "0.5.5";
+  sha256 = "sha256-BKBd1tJwV4VOjRnAx/spQy3LIXzujrO2SS5eA1uybNA=";
+  cargoSha256 = "sha256-dv04AXOnzizjq/qx3qy524ylQHgE4gIBgeYI+2IRTug=";
+
+  # Helpers
+  inherit (lib) optionals;
+  inherit (rustPlatform) buildRustPackage;
+  inherit (stdenv) isDarwin;
+
+  # Build inputs for all systems
+  standardInputs = [ openssl ];
+
+  # Build inputs for macOS only
+  darwinInputs = optionals isDarwin [ libiconv ]
+    ++ (with darwin.apple_sdk.frameworks; [ Security SystemConfiguration ]);
+in buildRustPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit owner repo sha256;
+    rev = "v${version}";
+  };
+
+  inherit cargoSha256;
+
+  buildInputs = standardInputs ++ darwinInputs;
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A package manager for WebAssembly modules";
+    homepage = "https://docs.wasmer.io/ecosystem/wapm";
+    license = with licenses; [ mit ];
+    maintainers = [ maintainers.lucperkins ];
+    platforms = platforms.all;
```

Comment:
```suggestion
```
rust defaults to platforms.linux ++ platforms.darwin which we do not need to repeat.